### ISSUE TITLE
Deprecated softmax and log_softmax function in PyTorch

### ISSUE THREAD

**Issue by ttpro1995** [OpenNMT/OpenNMT-py#632]
```
OpenNMT-py/onmt/modules/GlobalAttention.py:176: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.
  align_vectors = self.sm(align.view(batch*targetL, sourceL))

lib/python3.6/site-packages/torch/nn/modules/container.py:67: UserWarning: Implicit dimension choice for log_softmax has been deprecated. Change the call to include dim=X as an argument.
  input = module(input)

```

**Comment by Henry-E** [comment]
It appears to be fixed most softmax instances but not the copy generator yet https://github.com/OpenNMT/OpenNMT-py/blob/df0a0f27eccf569cbac8fd3ce7ff10f5e504cdcd/onmt/modules/CopyGenerator.py#L91
